docs/debugging.md

```diff
@@ -0,0 +1,130 @@
+# Debugging Project Verona
+
+We have built several features into the codebase to aid debugging.
+This document explains those features.
+If anything is unclear, please PR fixes or raise issues.
+
+We will split the rest of this document based on the various stages of running a Verona program
+
+* Compiler
+* Interpreter
+* Runtime
+
+## Debugging the compiler
+
+The simplist way to debug the compiler is to get it to dump all the phases of compiling every method:
+```
+veronac.exe --dump-path dmp region101.verona
+```
+This dumps all the stages of compiling each method into a sub-directory `dmp`.
+For each Class and Method, you should find:
+```
+  [Class].[Method].ast.txt
+  [Class].[Method].resolved-ast.txt
+  [Class].[Method].elaborated-ast.txt
+  [Class].[Method].wf-types.txt
+  [Class].[Method].ir.txt
+  [Class].[Method].liveness.txt
+  [Class].[Method].constraints.txt
+  [Class].[Method].infer.txt
+  [Class].[Method].solver.txt
+  [Class].[Method].substitution.txt
+  [Class].[Method].types.txt
+  [Class].[Method].typed-ir.txt
+  [Class].[Method].make-region-graph.txt
+  [Class].[Method].region-graph.txt
+```
+The order is the order of the phases in the compiler, and can be used to track where something has gone wrong.
+
+Individual, phases and results for particular methods can be output using other options to the compiler.
```

Comment:
I think the comma after "Individual" is misplaced. A more detailed note of how to use this would be neat. eg:

> Individual results can be printed to the standard output instead by passing the `--print=` flag to the compiler.
For example, the following command will print the intermediate representation for the `Main.main` method.

```
$ ./dist/veronac ../testsuite/demo/run-pass/bank1.verona --print=Main.main.ir
IR for Main.main:
  Basic block BB0:
    0 <- static Main
    1 <- integer 100
    2 <- call 0.example()
[...]

Reply:
Added